src/apps/activities/crud/activity_history.py

```diff
@@ -285,3 +285,11 @@ async def update_by_id(self, id_, **values):
         query = query.values(**values)
         query = query.returning(ActivityHistorySchema)
         await self._execute(query)
+
+    async def get_by_activity_id(
+        self, activity_id: uuid.UUID
+    ) -> ActivityHistorySchema:
+        schema = await self._get("id", activity_id)
+        if not schema:
+            raise ActivityHistoryDoeNotExist()
+        return schema
```

Comment:
The result of filtering is a list of activities with the same id but different version. You take random version here

Reply:
fixed